payments-core/src/main/java/com/stripe/android/model/ConsumerPaymentDetails.kt

```diff
@@ -29,6 +29,9 @@ data class ConsumerPaymentDetails internal constructor(
         val cvcCheck: CvcCheck
     ) : PaymentDetails(id, isDefault, Companion.type) {
 
+        val requiresCardDetailsRecollection: Boolean
+            get() = isExpired || cvcCheck.requiresRecollection
```

Comment:
If the card is expired, do we ask for a new expiry? Or does the payment go through?

Reply:
Yes, we will ask for the new expiry date. The UI for that is already in place, but I’m adding the logic for it in a future pull request. This one right here is focused on the CVC recollection.